modules/core/infrastructure/persistence/setup_test.go

```diff
@@ -33,23 +34,45 @@ func setupTest(t *testing.T) *testFixtures {
 	pool := testutils.NewPool(testutils.DbOpts(t.Name()))
 
 	ctx := context.Background()
+
+	// Setup application and run migrations first (outside the transaction)
+	app, err := testutils.SetupApplication(pool, modules.BuiltInModules...)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Run migrations first to create all tables including tenants
+	if err := app.Migrations().Run(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a test tenant outside transaction
+	tenant, err := testutils.CreateTestTenant(ctx, pool)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Now start the transaction
 	tx, err := pool.Begin(ctx)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	t.Cleanup(func() {
-		if err := tx.Commit(ctx); err != nil {
-			t.Fatal(err)
+		// Rollback instead of commit to ensure clean state
+		// This is safer as it ensures tests don't affect each other
```

Comment:
They already don't affect each other since each test is run in a separate db. Committing changes is useful to be able to introspect the db state after a failed test